ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -137,6 +141,10 @@ sk_sp<SkImage> RSkComponentImage::getLocalImageData(ImageSource source) {
     imageCacheData.expiryTime = (SkTime::GetMSecs() + DEFAULT_MAX_CACHE_EXPIRY_TIME);//convert min to millisecond 30 min *60 sec *1000
     RSkImageCacheManager::getImageCacheManagerInstance()->imageDataInsertInCache(source.uri.c_str(), imageCacheData);
   }
+  if(!hasToTriggerEvent_) {
+    imageEventEmitter_->onLoadStart();
```

Comment:
imageEventEmitter_->onLoadStart(); has been used in three places. Why can't all the Send events handling be moved to single API.

Will the below Style helps ?
emitImageEvents(imageEvent event){ 
  switch(event) {
  case onLoasStart:
    imageEventEmitter_->onLoadStart();
     break;
   case OnLoadFails:
      imageEventEmitter_->onError();
  imageEventEmitter_->onLoadEnd();
 break;
   case OnLoadcompletes:
      imageEventEmitter_->onLoad();
  imageEventEmitter_->onLoadEnd();
 break;
}
hasToTriggerEvent_ = true;
};